source/views/menus/carleton-menus.tsx

```diff
@@ -1,96 +1,73 @@
+import {RootStackParamList} from '../../navigation/types'
 
-export const CarletonBurtonMenuScreen = ({
-	navigation,
-}: TopLevelViewPropsType) => (
+export const CarletonBurtonMenuScreen = (): JSX.Element => (
 	<BonAppHostedMenu
 		cafe="burton"
 		loadingMessage={['Searching for Schiller…']}
 		name="Burton"
-		navigation={navigation}
 	/>
 )
-CarletonBurtonMenuScreen.navigationOptions = {
-	title: 'Burton',
-	tabBarIcon: TabBarIcon('menu'),
-}
 
-export const CarletonLDCMenuScreen = ({navigation}: TopLevelViewPropsType) => (
+export const CarletonLDCMenuScreen = (): JSX.Element => (
 	<BonAppHostedMenu
 		cafe="ldc"
 		loadingMessage={['Tracking down empty seats…']}
 		name="LDC"
-		navigation={navigation}
 	/>
 )
-CarletonLDCMenuScreen.navigationOptions = {
-	title: 'LDC',
-	tabBarIcon: TabBarIcon('menu'),
-}
 
-export const CarletonWeitzMenuScreen = ({
-	navigation,
-}: TopLevelViewPropsType) => (
+export const CarletonWeitzMenuScreen = (): JSX.Element => (
 	<BonAppHostedMenu
 		cafe="weitz"
 		loadingMessage={['Observing the artwork…', 'Previewing performances…']}
 		name="Weitz Center"
-		navigation={navigation}
 	/>
 )
-CarletonWeitzMenuScreen.navigationOptions = {
-	title: 'Weitz Center',
-	tabBarIcon: TabBarIcon('menu'),
-}
 
-export const CarletonSaylesMenuScreen = ({
-	navigation,
-}: TopLevelViewPropsType) => (
+export const CarletonSaylesMenuScreen = (): JSX.Element => (
 	<BonAppHostedMenu
 		cafe="sayles"
 		loadingMessage={['Engaging in people-watching…', 'Checking the mail…']}
 		name="Sayles Hill"
-		navigation={navigation}
 	/>
 )
-CarletonSaylesMenuScreen.navigationOptions = {
-	title: 'Sayles Hill',
-	tabBarIcon: TabBarIcon('menu'),
-}
 
-type Props = TopLevelViewPropsType
+export function CarletonCafeIndex(): JSX.Element {
+	let navigation = useNavigation()
 
-export function CarletonCafeIndex(props: Props) {
-	let carletonCafes = [
-		{id: 'CarletonBurtonMenuView', title: 'Burton'},
-		{id: 'CarletonLDCMenuView', title: 'LDC'},
-		{id: 'CarletonWeitzMenuView', title: 'Weitz Center'},
-		{id: 'CarletonSaylesMenuView', title: 'Sayles Hill'},
+	let carletonCafes: Array<{id: keyof RootStackParamList; title: string}> = [
+		{id: 'CarletonBurtonMenu', title: 'Burton'},
+		{id: 'CarletonLDCMenu', title: 'LDC'},
+		{id: 'CarletonWeitzMenu', title: 'Weitz Center'},
+		{id: 'CarletonSaylesMenu', title: 'Sayles Hill'},
 	]
 
 	return (
 		<ScrollView style={styles.container}>
-			{carletonCafes.map((loc: {id: string; title: string}, i, collection) => (
-				<View key={i}>
-					<ListRow
-						arrowPosition="center"
-						onPress={() => props.navigation.navigate(loc.id)}
-					>
-						<Row alignItems="center">
-							<Title style={styles.rowText}>{loc.title}</Title>
-						</Row>
-					</ListRow>
-					{i < collection.length - 1 ? (
-						<ListSeparator spacing={{left: 15}} />
-					) : null}
-				</View>
-			))}
+			{carletonCafes.map(
+				(loc: {id: keyof RootStackParamList; title: string}, i, collection) => (
```

Comment:
Do we need still this type annotation? Or can TS infer it?

Reply:
Looks like we don't! Good catch